### PR TITLE
Dto validation

### PR DESCRIPTION
**Pull request by ColeSamuelsen23**
Added tests to check for DTO data annotation behaviour on NewPostDTO model.

### CODE REVIEW

File: Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs [hunk 1]
```diff
@@ -113,15 +115,38 @@ public async Task PostAsync_InvalidPost_BadRequest()
...
+            }
+
+            //Create New Post DTO objects that will be sent to the API.
+
+            NewPostDto invalidPostTooLong = new() { Content = longPost.ToString() };
+            NewPostDto invalidPostNoContent = new() { Content = "" };
+
+            //  Serializes request object into json format.
+
+            StringContent invalidStringTooMuchContent = new(System.Text.Json.JsonSerializer.Serialize(invalidPostTooLong), Encoding.UTF8, "application/json");
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs:134]: use using directives instead of full names

**Review by escalonn — CHANGES_REQUESTED** [review]

File: Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs (hunk 1 above)
Comment (ColeSamuelsen23) [reply Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs]: Fixed it in every place  I could see, I'll do that for the other files too as I go along

File: Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs [hunk 2]
```diff
@@ -111,17 +116,40 @@ public async Task PostAsync_InvalidPost_BadRequest()
...
 
             // Assert
-            Assert.Equal(System.Net.HttpStatusCode.BadRequest, response.StatusCode);
+            // Ensures that the valid request was created, and that posts that are too long or too short are rejected.
+
+            Assert.Equal(HttpStatusCode.BadRequest, invalidresponseTooMuchContent.StatusCode);
+            Assert.Equal(HttpStatusCode.BadRequest, invalidResponseNoContent.StatusCode);
+
+
+
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs:152]: what's with all the blank lines

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs [hunk 3]
```diff
@@ -100,7 +105,7 @@ public async Task<IActionResult> PutAsync(int id, Post post)
         [ProducesResponseType(StatusCodes.Status201Created)]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
-        public async Task<IActionResult> PostAsync(NewPostDto postModel)
+        public async Task<IActionResult> PostAsync([Required]NewPostDto postModel)
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:108]: i think this is superfluous, but see what happens if you send in the body `null`, or nothing (which i don't think is valid JSON) with and without this attribute.
if it's superfluous, you could mark the code smells as false positive

**Review by escalonn — CHANGES_REQUESTED** [review]
i'm annoyed i didn't notice this before, but the folder `DTOs` and the namespace `DTOs` should both have been `Dtos`, the same naming convention that applies to type names applies to those.

File: Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs (hunk 2 above)
Comment (ColeSamuelsen23) [reply Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/PostCreation.cs:152]: I changed it so comments are right above what they are documenting, also removed a couple other unnecessary spaces. Fixed

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs (hunk 3 above)
Comment (ColeSamuelsen23) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:108]: I sent in the body as actual null {Content: null} and also just an empty object and both were giving me bad request with or without [Required] and that's what I want so Im going to call that code smell a false positive and be done with it.

**Review by escalonn — APPROVED** [review]

**Review by nosyrbllewe — APPROVED** [review]

**Pull request merged**